### ISSUE TITLE
Proposal: Add a callback when a cache item expires

### ISSUE THREAD

**Issue by Vlad-Zhukov** [planttheidea/moize#59]
There is currently no way to know that a cache item has expired, it just happens and you can't do anything with it. I propose to add a callback function that is called with a `key` inside of the `expireAfter` method.

I see 2 public API variants:

__1. As a `moize` options argument__

```js
const fn = () => {};
const moized = moize(fn, {
  maxAge: 10000,
  onExpiration(key) {
  // ...
  },
});
```
It's the simpliest way to do that, I guess, but it's hard to control the callback after initialization.

__2. As a subscription to events__

It's also possible to use one of observer patters, see [Comparison between different Observer Pattern implementations](https://github.com/millermedeiros/js-signals/wiki/Comparison-between-different-Observer-Pattern-implementations)

```js
const fn = () => {};
const moized = moize(fn, {
  maxAge: 10000,
});

moized.onExpiration.add((key) => {
  // ...
});
```

What do you think?

**Comment by planttheidea** [comment]
Sure, shouldn't be too crazy to do the first option. Not a fan of the second option, would involve a lot of under the hood changes that could impact code size and performance.

Is the key the only parameter you would want passed? Maybe the cache as a second parameter?

**Comment by Vlad-Zhukov** [comment]
I've no idea, why would one need it for? Most of the methods are accessible anyway.

Also I would like to contribute this feature if you don't mind 😃

**Comment by planttheidea** [comment]
Haha fair enough, just asking the question. And absolutely! PRs are always welcome. 😁

**Comment by planttheidea** [comment]
Released in `3.4.0`. Thanks again for the PR!
